pkg/kubernetes/scripts/kube-client-cockpit.js

```diff
@@ -1061,11 +1061,17 @@
 
                 var env_p = CockpitEnvironment()
                     .then(function(result) {
-                        var value = result["REGISTRY_HOST"];
-                        if (value) {
-                            settings.registry.host = value;
+                        var regHost = result["REGISTRY_HOST"];
+                        if (regHost) {
+                            settings.registry.host = regHost;
                             settings.registry.host_explicit = true;
                         }
+                        var kubehost = result["COCKPIT_KUBE_URL"];
```

Comment:
no, COCKPIT_KUBE_URL is the public URL for the container, so that isn't what you want. KUBERNETES_SERVICE_HOST is where the api server is. But that may be the private IP and not accessable from outside the cluster. Really what we want is the domain part of OPENSHIFT_OAUTH_PROVIDER_URL. But that is not available unless it is running in container mode.